media-libs/speexdsp/speexdsp-1.2_rc3.ebuild

```diff
@@ -0,0 +1,58 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+inherit autotools flag-o-matic multilib-minimal
+
+MY_P=${P/_} ; MY_P=${MY_P/_p/.}
+
+DESCRIPTION="Audio compression format designed for speech -- DSP"
+HOMEPAGE="http://www.speex.org/"
+SRC_URI="http://downloads.xiph.org/releases/speex/${MY_P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~sh ~sparc ~x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos ~sparc-solaris ~x86-solaris"
+IUSE="fixed-point fftw mkl static-libs cpu_flags_x86_sse cpu_flags_x86_sse2 neon"
+
+REQUIRED_USE="?? ( fixed-point fftw mkl )"
+
+RDEPEND="
+	fftw? ( sci-libs/fftw:3.0 )
+	mkl? ( sci-libs/mkl )"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig"
+
+S=${WORKDIR}/${MY_P}
+
+PATCHES=( "${FILESDIR}/${P}-configure.patch" )
+
+src_prepare() {
+	default
+
+	sed -i \
+		-e 's:noinst_PROGRAMS:check_PROGRAMS:' \
+		libspeexdsp/Makefile.am || die
+
+	eautoreconf
+}
+
+multilib_src_configure() {
+	append-lfs-flags
+
+	ECONF_SOURCE="${S}" econf \
+		$(use_enable static-libs static) \
+		$(use_enable cpu_flags_x86_sse sse) \
+		$(use_enable cpu_flags_x86_sse2 sse2) \
+		$(use_enable neon) \
+		$(use_enable fixed-point) \
+		$(usex fftw '--with-fft=gpl-fftw3' '') \
+		$(usex mkl '--with-fft=proprietary-intel-mkl' '')
```

Comment:
The `--with-fft` configure flag takes a couple of different options.
Using `REQUIRED_USE` and `usex` like this made the most sense to me -- but I'm wondering if there's a better way, specifically using `use_with`.

The values that it takes are either:
`kiss` -- default fixed-point FFT (always used with `--enable-fixed-point`)
`smallft` -- default floating point FFT
`gpl-fftw3` -- Use FFTW's FFT
`proprietary-intel-mkl` -- Use the Intel MKL FFT